mythril/mythril.py

```diff
@@ -118,40 +119,69 @@ def _init_mythril_dir(self):
-
-        if not os.path.exists(config_path):
-            logging.info("No config file found. Creating default: " + config_path)
-
-            config = ConfigParser(allow_no_value=True)
-            config.optionxform = str
-            config.add_section('defaults')
-            config.set('defaults', "#Default chaindata locations:")
-            config.set('defaults', "#– Mac: ~/Library/Ethereum/geth/chaindata")
-            config.set('defaults', "#– Linux: ~/.ethereum/geth/chaindata")
-            config.set('defaults', "#– Windows: %USERPROFILE%\\AppData\\Roaming\\Ethereum\\geth\\chaindata")
-            config.set('defaults', 'leveldb_dir', fallback_dir)
-            with codecs.open(config_path, 'w', 'utf-8') as fp:
-                config.write(fp)
+            leveldb_fallback_dir = os.path.join(leveldb_fallback_dir, ".ethereum")
+        leveldb_fallback_dir = os.path.join(leveldb_fallback_dir, "geth", "chaindata")
+
+        if not os.path.exists(self.config_path):
+            logging.info("No config file found. Creating default: " + self.config_path)
+            open(self.config_path, 'a').close()
 
         config = ConfigParser(allow_no_value=True)
         config.optionxform = str
-        config.read(config_path, 'utf-8')
-        leveldb_dir = config.get('defaults', 'leveldb_dir', fallback=fallback_dir)
+        config.read(self.config_path, 'utf-8')
+        if 'defaults' not in config.sections() :
+            config = self._add_default_options(config)
+
+        if not config.has_option('defaults', 'leveldb_dir'):
+            config = self._add_leveldb_option(config, leveldb_fallback_dir)
+
+        if not config.has_option('defaults', 'dynamic_loading'):
+            config = self._add_dynamic_loading_option(config)
+
+        with codecs.open(self.config_path, 'w', 'utf-8') as fp:
+            config.write(fp)
+
+        config.read(self.config_path, 'utf-8')
+        leveldb_dir = config.get('defaults', 'leveldb_dir', fallback=leveldb_fallback_dir)
         return os.path.expanduser(leveldb_dir)
 
+    @staticmethod
+    def _add_default_options(config):
+        config.add_section('defaults')
+        return config
+
+    @staticmethod
+    def _add_leveldb_option(config, leveldb_fallback_dir):
+        config.set('defaults', "#Default chaindata locations:")
+        config.set('defaults', "#– Mac: ~/Library/Ethereum/geth/chaindata")
+        config.set('defaults', "#– Linux: ~/.ethereum/geth/chaindata")
+        config.set('defaults', "#– Windows: %USERPROFILE%\\AppData\\Roaming\\Ethereum\\geth\\chaindata")
+        config.set('defaults', 'leveldb_dir', leveldb_fallback_dir)
+        return config
+
+    @staticmethod
+    def _add_dynamic_loading_option(config):
+        config.set('defaults', '#– To connect to Infura use dynamic_loading: infura')
+        config.set('defaults', '#– To connect to Ipc use dynamic_loading: ipc')
+        config.set('defaults', '#– To connect to Rpc use '
```

Comment:
Should we be adding comments for every option or can we just mention that properly in the wiki?

Reply:
I think it doesn't hurt to also have them here